File: exercise/java/day25/src/main/java/timecapsule/Browser.java
package timecapsule;

import java.awt.Desktop;
import java.io.File;
import java.io.IOException;

public class Browser {
    public static void open(String filePath) {
        try {
            var file = new File(filePath);
            if (file.exists() && Desktop.isDesktopSupported()) {
                Desktop.getDesktop().browse(file.toURI());
            } else {
                throw new IOException("File not found or desktop environment not supported.");
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: exercise/kotlin/day25/src/main/kotlin/timecapsule/Browser.kt
package timecapsule

import java.awt.Desktop.*
import java.io.File

object Browser {
    fun open(filePath: String) = if (isDesktopSupported() && getDesktop().isSupported(Action.OPEN))
        getDesktop().open(File(filePath))
    else println("Opening files is not supported on this system.")
}